### PR TITLE
Merge-template-2.9 & mirbase fixes

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,26 +57,8 @@ You can find numerous talks on the nf-core events page from various topics inclu
...
-
-```csv
-sample,fastq_1
-CONTROL_REP1,AEG588A1_S1_L002_001.fastq.gz
-```
-
-Each row represents a fastq file (single-end).
-
--->
-
```
Comment (grst) [inline README.md:75]: Would be good to actually show how the samplesheet needs to look like

File: README.md [hunk 2]
```diff
@@ -57,26 +57,8 @@ You can find numerous talks on the nf-core events page from various topics inclu
...
 Now, you can run the pipeline using:
 
-<!-- TODO nf-core: update the following command to include all required parameters for a minimal example -->
-
 ```bash
 nextflow run nf-core/smrnaseq \
    -profile <docker/singularity/.../institute> \
```
Comment (grst) [inline README.md:64]: something's wrong with the indentation in the next lines

File: nextflow.config [hunk 3]
```diff
@@ -89,6 +84,20 @@ params {
     max_cpus                    = 16
     max_time                    = '240.h'
 
+    // Schema validation default options
+    validationFailUnrecognisedParams = false
+    validationLenientMode            = false
+    validationSchemaIgnoreParams     = 'genomes'
+    validationShowHiddenParams       = false
+    validate_params                  = true
```
Comment (grst) [inline nextflow.config:92]:
```suggestion
```

(duplication...)

File: nextflow.config [hunk 4]
```diff
@@ -255,8 +277,8 @@ manifest {
     homePage        = 'https://github.com/nf-core/smrnaseq'
     description     = """Small RNA-Seq Best Practice Analysis Pipeline."""
     mainScript      = 'main.nf'
-    nextflowVersion = '!>=22.10.1'
-    version         = '2.2.2'
+    nextflowVersion = '!>=23.04.0'
+    version = '2.2.1dev'
```
Comment (grst) [inline nextflow.config:281]: Shouldn't this be `2.3dev` of `2.2.2dev` as `2.2.1` is already released?

**Review by grst — CHANGES_REQUESTED** [review]
Found a couple of minor things

**Review by grst — APPROVED** [review]
LGTM now!

**Review by christopher-mohr — COMMENTED** [review]
As just discussed LGTM, we just need to adapt to the new `samtools` subworkflows.

File: subworkflows/local/mirna_quant.nf [hunk 5]
```diff
@@ -15,8 +15,8 @@ include {   BOWTIE_MAP_SEQ  as BOWTIE_MAP_MATURE
             BOWTIE_MAP_SEQ  as BOWTIE_MAP_HAIRPIN
             BOWTIE_MAP_SEQ  as BOWTIE_MAP_SEQCLUSTER } from '../../modules/local/bowtie_map_mirna'
 
-include {   BAM_SORT_SAMTOOLS as BAM_STATS_MATURE
-            BAM_SORT_SAMTOOLS as BAM_STATS_HAIRPIN   } from '../nf-core/bam_sort_samtools'
+include {   BAM_SORT_STATS_SAMTOOLS as BAM_STATS_MATURE
```
Comment (christopher-mohr) [inline subworkflows/local/mirna_quant.nf:18]: The second argument of these calls below needs to be a map like `[meta, fasta]. Maybe the easiest solution would be to also adapt the module `format_fasta_mirna` to also accept and return a map of `[meta, fasta]`.

File: workflows/smrnaseq.nf [hunk 6]
```diff
@@ -96,7 +102,7 @@ workflow SMRNASEQ {
     // SUBWORKFLOW: Read in samplesheet, validate and stage input files
     //
     INPUT_CHECK (
-        ch_input
+        file(params.input)
```
Comment (christopher-mohr) [inline workflows/smrnaseq.nf:105]: Shouldn't this still be `ch_input` ?

**Review by christopher-mohr — APPROVED** [review]
One comment, apart from that LGTM!

**Pull request merged**